app/(api)/_utils/response/DuplicateError.ts

```diff
@@ -0,0 +1,9 @@
+import HttpError from './HttpError';
+
+export default class DuplicateError extends HttpError {
+  constructor(message: string) {
+    super(message);
+    this.name = 'DuplicateError';
+    this.status = 409;
```

Comment:
nice :)

Reply:
if u wanna improve the custom error situation feel free as long as it doesn't break too much of Win's code lol